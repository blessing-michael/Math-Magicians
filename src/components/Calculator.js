import React from 'react';
import CalculatorButtons from './calculatorButtons';
import CalculatorDisplay from './CalculatorDisplay';
import '../styles/style.css';

const Calculator = () => (
  <div className="calculator-container">
    <CalculatorDisplay />
    <CalculatorButtons />

  </div>
);

export default Calculator;
