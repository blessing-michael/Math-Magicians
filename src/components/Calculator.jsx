import React, { useState } from 'react';
import '../styles/style.css';
import calculate from '../utils/calculate';
import CalculatorDisplay from './CalculatorDisplay';
import CalculatorButtons from './CalculatorButtons';

const Calculator = () => {
  const [calculatorData, setCalculatorData] = useState({
    total: null,
    next: null,
    operation: null,
  });

  const handleClickedButton = (buttonName) => {
    const updatedData = calculate(calculatorData, buttonName);
    setCalculatorData(updatedData);
  };

  return (
    <div className="calculator-container">
      <CalculatorDisplay displayValue={calculatorData.next || calculatorData.total || '0'} />
      <CalculatorButtons handleClickedButton={handleClickedButton} />
    </div>
  );
};

export default Calculator;
