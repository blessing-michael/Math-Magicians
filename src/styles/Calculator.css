.calc-container {
  width: 480px;
  margin: 0 auto;
  display: flex;
  flex-direction: column;
}

h2 {
  font-family: sans-serif;
  font-size: 1rem;
  text-align: center;
}

.all-btns {
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(100px, 1fr));
}

.all-btns .grid-item:nth-child(4n),
.all-btns .grid-item:nth-child(19) {
  background-color: hsl(27, 87%, 63%);
}

input[type="text"] {
  padding: 1rem;
  text-align: right;
  font-family: sans-serif;
  font-size: 2rem;
  background-color: hsl(233, 4%, 58%);
}

input[type="text"]::placeholder {
  color: whitesmoke;
}

button {
  padding: 1rem;
  outline: none;
  border: 0.5px solid hsl(231, 3%, 57%);
  font-family: sans-serif;
  font-size: 1rem;
}

.span-2 {
  grid-column: span 2;
}
