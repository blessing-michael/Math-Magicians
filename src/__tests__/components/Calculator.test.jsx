import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import Calculator from '../../components/Calculator';

describe('Calculator', () => {
  test('renders correctly', () => {
    const { container } = render(<Calculator />);
    expect(container).toMatchSnapshot();
  });

  test('displays correct value on button click', () => {
    const { getByRole, getByText } = render(<Calculator />);
    const displayElement = getByRole('textbox');

    // Simulate button clicks and assert display value updates accordingly
    fireEvent.click(getByText('1'));
    expect(displayElement).toHaveDisplayValue('1');

    fireEvent.click(getByText('+'));
    expect(displayElement).toHaveDisplayValue('1');

    fireEvent.click(getByText('5'));
    expect(displayElement).toHaveDisplayValue('5');

    fireEvent.click(getByText('='));
    expect(displayElement).toHaveDisplayValue('6');
  });
});
