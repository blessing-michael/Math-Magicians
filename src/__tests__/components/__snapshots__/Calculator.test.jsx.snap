// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator renders correctly 1`] = `
<div>
  <div
    class="calc-container"
  >
    <h2
      class="intro-text"
    >
      Your Perfect Math Calculator
    </h2>
    <input
      disabled=""
      style="color: whitesmoke;"
      type="text"
      value="0"
    />
    <div
      class="all-btns"
    >
      <button
        class="grid-item undefined"
        type="button"
      >
        AC
      </button>
      <button
        class="grid-item undefined"
        type="button"
      >
        +/-
      </button>
      <button
        class="grid-item undefined"
        type="button"
      >
        %
      </button>
      <button
        class="grid-item undefined"
        type="button"
      >
        รท
      </button>
      <button
        class="grid-item undefined"
        type="button"
      >
        7
      </button>
      <button
        class="grid-item undefined"
        type="button"
      >
        8
      </button>
      <button
        class="grid-item undefined"
        type="button"
      >
        9
      </button>
      <button
        class="grid-item undefined"
        type="button"
      >
        x
      </button>
      <button
        class="grid-item undefined"
        type="button"
      >
        4
      </button>
      <button
        class="grid-item undefined"
        type="button"
      >
        5
      </button>
      <button
        class="grid-item undefined"
        type="button"
      >
        6
      </button>
      <button
        class="grid-item undefined"
        type="button"
      >
        -
      </button>
      <button
        class="grid-item undefined"
        type="button"
      >
        1
      </button>
      <button
        class="grid-item undefined"
        type="button"
      >
        2
      </button>
      <button
        class="grid-item undefined"
        type="button"
      >
        3
      </button>
      <button
        class="grid-item undefined"
        type="button"
      >
        +
      </button>
      <button
        class="grid-item span-2"
        type="button"
      >
        0
      </button>
      <button
        class="grid-item undefined"
        type="button"
      >
        .
      </button>
      <button
        class="grid-item undefined"
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
